File: testing/example_2000_IDN_points.py
import os
import boto3
import json
import uuid


client = boto3.client('lambda')

# load geojson test points
# source: q
cwd = os.path.dirname(os.path.abspath(__file__))
with open(os.path.join(cwd, 'idn_points.geojson')) as thefile:
    data = json.load(thefile)

for feature in data['features']:

    lon, lat = feature['geometry']['coordinates']

    # build an out dir that includes lat and lon and random GUID
    # will probably change this in the future, but nice right now
    # to have lat/lon in there for debugging
    out_dir = '{}_{}_{}'.format(round(lat, 5), round(lon, 5), str(uuid.uuid4()))

    event = {
    'queryStringParameters': {
        'out_dir': out_dir,
        'lat': lat,
        'lon': lon
        }
    }

    client.invoke(
        FunctionName='palm-risk-poc-dev-mill',
        InvocationType='Event',
        Payload=json.dumps(event)
        )

File: testing/example_2000_IDN_points.js
var AWS = require('aws-sdk');
const uuidv4 = require('uuid/v4');

AWS.config.update({region: 'us-east-1'});

var lambda = new AWS.Lambda();

var fs = require('fs');
var geojson = JSON.parse(fs.readFileSync(__dirname + '/' + 'idn_points.geojson', 'utf8'));

var features = geojson.features.slice(0,1)

for (var i = 0; i < features.length; i++) {

  var lon = features[i].geometry.coordinates[0]
  var lat = features[i].geometry.coordinates[1]

  var out_dir = lat + '_' + lon + '_' + uuidv4()

  var payload = {
    "queryStringParameters": {
        "out_dir": out_dir,
        "lat": lat,
        "lon": lon
        }
    }

  var params = {
    FunctionName: 'palm-risk-poc-dev-mill',
    Payload: JSON.stringify(payload)
  }

  lambda.invoke(params, function(err, data) {
    if (err) console.log(err, err.stack); // an error occurred
  });

}
